.github/workflows/integration_test.yml

```diff
@@ -0,0 +1,177 @@
+name: Integration tests
```

Comment:
@lionel-nj Would you be able to add an INTEGRATION_TESTS.md (or whatever you want to name it) that documents the workflow at a high level, including what the GitHub Action script does and it's relation to the Java application? I think this would help clarify expected behavior here, expected as related to the `matrix`.

Reply:
IIRC you had a diagram describing this at one point - it would be great to include that too.

Reply:
Absoluetly! Will add this doc!

Reply:
@barbeau The documentation was updated in the latest commit.